File: Q63/Q63.py
# Implement quicksort

def quicksort(arr):
    if len(arr) <= 1:
        return arr
    pivot = arr[len(arr) // 2]
    left = [x for x in arr if x < pivot]
    middle = [x for x in arr if x == pivot]
    right = [x for x in arr if x > pivot]
    return quicksort(left) + middle + quicksort(right)

array = [10, 7, 8, 9, 1, 5]
sorted_array = quicksort(array)
print("Sorted Array:", sorted_array)

File: Q63/Q63.js
//  Implement quicksort

function quicksort(arr) {
  if (arr.length <= 1) {
    return arr;
  }
  const pivot = arr[Math.floor(arr.length / 2)];
  const left = arr.filter((x) => x < pivot);
  const middle = arr.filter((x) => x === pivot);
  const right = arr.filter((x) => x > pivot);
  return [...quicksort(left), ...middle, ...quicksort(right)];
}

const array = [10, 7, 8, 9, 1, 5];
const sortedArray = quicksort(array);
console.log("Sorted Array:", sortedArray);
